File: ext/owl/router/routerinterface.zep.c

#ifdef HAVE_CONFIG_H
#include "../../ext_config.h"
#endif

#include <php.h>
#include "../../php_ext.h"
#include "../../ext.h"

#include <Zend/zend_exceptions.h>

#include "kernel/main.h"


ZEPHIR_INIT_CLASS(Owl_Router_RouterInterface) {

	ZEPHIR_REGISTER_INTERFACE(Owl\\Router, RouterInterface, owl, router_routerinterface, owl_router_routerinterface_method_entry);

	return SUCCESS;

}

ZEPHIR_DOC_METHOD(Owl_Router_RouterInterface, matchRequest);

ZEPHIR_DOC_METHOD(Owl_Router_RouterInterface, match);

File: ide/0.0.1/Owl/router/RouterInterface.zep.php
<?php

namespace Owl\Router;

interface RouterInterface
{

    /**
     * @param mixed $request 
     */
	public function matchRequest(Owl\Http\RequestInterface $request);

    /**
     * @param string $uri 
     * @param string $method 
     * @return object|bool 
     */
	public function match($uri, $method = Request::GET);

}
